### PR TITLE
Align RoomSummaryCard styles with Figma

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/element-hq/wat-internal/issues/166
~Requires https://github.com/matrix-org/matrix-react-sdk/pull/12796~
~Requires https://github.com/matrix-org/matrix-react-sdk/pull/12843~

Review commit-by-commit

Designs https://www.figma.com/design/gpxHFDTNK796n84r1ZxYG2/Room-Header-%26-Details-(2024)?node-id=0-1&t=NCusToRsR4Vju4OT-0

image

### CODE REVIEW

**Review by richvdh — CHANGES_REQUESTED** [review]
Generally lgtm. A few nits about documentation and others.

Thanks for making the effort to help review by breaking it down into smaller commits - that definitely helped. Though I would say it feels like it could have been two PRs - one to move the extensions to a new tab, and another to do the other rearrangements to the summary card.

File: src/components/views/right_panel/ExtensionsCard.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            managers.openNoManagerDialog();
+        } else {
+            // noinspection JSIgnoredPromiseFromCall
+            managers.getPrimaryManager()?.open(room);
+        }
+    };
+
+    const header = (
+        <Button size="sm" onClick={onManageIntegrations} kind="secondary" Icon={PlusIcon}>
+            {_t("right_panel|add_integrations")}
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/right_panel/ExtensionsCard.tsx:218]: The designs don't have such a state anymore

File: src/components/views/right_panel/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import { Flex } from "../../utils/Flex";
+
+interface Props {
+    Icon: ComponentType<React.SVGAttributes<SVGElement>>;
+    title: string;
+    description: string;
+}
+
+const EmptyState: React.FC<Props> = ({ Icon, title, description }) => {
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/right_panel/EmptyState.tsx]: Sorry this landed in a separate PR already

File: res/css/views/right_panel/_EmptyState.pcss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        width: 56px;
+        height: 56px;
+        box-sizing: border-box;
+        border-radius: 8px;
+        padding: var(--cpd-space-3x);
+        background-color: $panel-actions;
+    }
+
+    &::before {
+        /* Bloom using magic numbers directly out of Figma */
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: Sorry this landed in a separate PR already

File: res/css/views/right_panel/_EmptyState.pcss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        display: block;
+        position: absolute;
+        z-index: -1;
+        opacity: 0.6;
+        width: 642px;
+        height: 775px;
+        right: -253.77px;
+        top: 0;
+        background: radial-gradient(49.95% 49.95% at 50% 50%, rgba(13, 189, 139, 0.12) 0%, rgba(18, 115, 235, 0) 100%);
+        transform: rotate(-89.69deg);
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: You'd have to ask design, these are the parameters which they specified in Figma

[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: (and this already landed)

File: res/css/views/right_panel/_EmptyState.pcss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    &::before {
+        /* Bloom using magic numbers directly out of Figma */
+        content: "";
+        display: block;
+        position: absolute;
+        z-index: -1;
+        opacity: 0.6;
+        width: 642px;
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: This already landed and is what is specified in the designs

File: res/css/views/right_panel/_EmptyState.pcss [hunk 6]
```diff
@@ -32,10 +32,8 @@ limitations under the License.
     &::before {
         /* Bloom using magic numbers directly out of Figma */
         content: "";
-        display: block;
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: Yeah sorry this PR ended up a mess as I ended up splitting out EmptyState as another PR needed it and then forgot to rebase here. Do keep meaning to try `git-absorb`, thanks

File: src/components/structures/RightPanel.tsx [hunk 7]
```diff
@@ -306,6 +307,12 @@ export default class RightPanel extends React.Component<Props, IState> {
                 }
                 break;
 
+            case RightPanelPhases.Extensions:
+                if (!!this.props.room) {
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/structures/RightPanel.tsx:311]: Yeah went for consistency here

File: res/css/views/right_panel/_EmptyState.pcss (hunk 4 above)
[the remark this answers is not in the document]
Comment (richvdh) [reply res/css/views/right_panel/_EmptyState.pcss]: why am I wasting hours of my time reviewing things that have already landed

[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/views/right_panel/_EmptyState.pcss]: https://github.com/matrix-org/matrix-react-sdk/pull/12793#discussion_r1689544665 has the explanation

**Comment by t3chguy** [comment]
`useIsVideoRoom` split out into https://github.com/matrix-org/matrix-react-sdk/pull/12843

**Comment by t3chguy** [comment]
Extensions tab split out into https://github.com/matrix-org/matrix-react-sdk/pull/12844

**Review by dbkr — APPROVED** [review]

**Pull request merged**